src/components/user-profile/index.js

```diff
@@ -1,30 +1,23 @@
 import { useState, useEffect } from 'react';
 import { userContext } from '@store/user/user-context';
 import classNames from '@components/user-profile/user-profile.module.scss';
-import { getUserSelf } from '../../helper-functions/action-handlers';
 
 const UserProfile = () => {
   const [isLoading, setIsLoading] = useState(true);
   const {
     user,
-    setUser,
     isSuperUser,
-    setIsSuperUser,
     isSuperUserMode,
     setIsSuperUserMode,
+    verifySuperUser,
   } = userContext();
 
   useEffect(() => {
-    async function getUserProfile() {
-      const { data } = await getUserSelf();
+    (async () => {
+      await verifySuperUser();
       setIsLoading(false);
-      setUser(data);
-      if (data) {
-        setIsSuperUser(data.roles.super_user);
-      }
-    }
-    getUserProfile();
-  }, [setUser, setIsSuperUser]);
+    })();
+  }, [verifySuperUser]);
```

Comment:
This should ideally not be the responsibility of the component. It should exist in an API util, or data layer.

Reply:
Agreed. I will create a new ticket to move this into separate API util or data layer.